src/autopas/utils/RaplMeter.cpp

```diff
@@ -0,0 +1,167 @@
+/**
+ * @file RaplMeter.cpp
+ * @author Vincent Fischer
+ * @date 09.11.2021
+ */
+
+#include "autopas/utils/RaplMeter.h"
+
+#include <linux/perf_event.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+#include <cmath>
+#include <cstring>
+#include <sstream>
+
+#include "autopas/utils/ExceptionHandler.h"
+#include "autopas/utils/logging/Logger.h"
+
+namespace autopas::utils {
+
+int RaplMeter::open_perf_event(int type, int config) {
+  struct perf_event_attr attr;
+  memset(&attr, 0, sizeof(attr));
+  attr.type = type;
+  attr.config = config;
+  if (config == 0) {
+    return -1;
+  }
+
+  int fd = syscall(__NR_perf_event_open, &attr, -1 /*PID*/, 0 /*CPU*/, -1 /*GROUP FD*/, 0 /*FLAGS*/);
+  if (fd < 0) {
+    if (errno == EACCES) {
+      throw ExceptionHandler::AutoPasException("Failed to open perf event: Permission denied");
+    }
+    throw ExceptionHandler::AutoPasException("Failed to open perf event");
+  }
+  // TODO: Error handling
```

Comment:
open TODO :)

Reply:
Improved this a bit. Current behavior if an error occurs during energy measurement:
If energy is selected as tuning metric -> crash immediately
if time is selected -> log a warning and stop measuring energy in following iterations

Reply:
sounds very reasonable